src/infra/core/src/ingest/fetch_service.rs

```diff
@@ -24,29 +25,113 @@ use super::*;
+
+#[derive(Debug, Clone)]
+pub struct SourceConfig {
+    /// Target number of records after which we will stop consuming from the
+    /// resumable source and commit data, leaving the rest for the next
+    /// iteration. This ensures that one data slice doesn't become too big.
+    pub target_records_per_slice: u64,
+}
+
+impl Default for SourceConfig {
+    fn default() -> Self {
+        Self {
+            target_records_per_slice: 10_000,
+        }
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone)]
+pub struct MqttSourceConfig {
+    /// Time in milliseconds to wait for MQTT broker to send us some data after
+    /// which we will consider that we have "caught up" and end the polling
+    /// loop.
+    pub broker_idle_timeout_ms: u64,
+}
+
+impl Default for MqttSourceConfig {
+    fn default() -> Self {
+        Self {
+            broker_idle_timeout_ms: 1_000,
+        }
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone)]
+pub struct EthereumSourceConfig {
+    /// Default RPC endpoints to use if source does not specify one explicitly.
+    pub rpc_endpoints: Vec<EthRpcEndpoint>,
+    /// Default number of blocks to scan within one query to `eth_getLogs` RPC
+    /// endpoint.
+    pub get_logs_block_stride: u64,
+    // TODO: Consider replacing this with logic that upon encountering an error still commits the
+    // progress made before it
+    /// Forces iteration to stop after the specified number of blocks were
+    /// scanned even if we didn't reach the target record number. This is useful
+    /// to not lose a lot of scanning progress in case of an RPC error.
+    pub commit_after_blocks_scanned: u64,
+}
+
+impl Default for EthereumSourceConfig {
+    fn default() -> Self {
+        Self {
+            rpc_endpoints: Vec::new(),
+            get_logs_block_stride: 100_000,
+            commit_after_blocks_scanned: 1_000_000,
+        }
+    }
+}
+
+impl EthereumSourceConfig {
+    pub fn get_endpoint_by_chain_id(&self, chain_id: u64) -> Option<&EthRpcEndpoint> {
+        self.rpc_endpoints.iter().find(|e| e.chain_id == chain_id)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct EthRpcEndpoint {
+    pub chain_id: u64,
+    pub chain_name: String,
+    pub node_url: Url,
+}
 
 ////////////////////////////////////////////////////////////////////////////////
 
 pub struct FetchService {
     container_runtime: Arc<ContainerRuntime>,
-    container_log_dir: PathBuf,
+    run_info_dir: PathBuf,
+    source_config: Arc<SourceConfig>,
+    mqtt_source_config: Arc<MqttSourceConfig>,
+    eth_source_config: Arc<EthereumSourceConfig>,
 }
 
 ////////////////////////////////////////////////////////////////////////////////
 
+#[dill::component(pub)]
 // TODO: Split this service apart into pluggable protocol implementations
 impl FetchService {
```

Comment:
Consider splitting the service on multiple. This could be a facade that decodes the source and coordinates detailed fetch services.